gflownet.py

```diff
@@ -700,23 +705,41 @@ def sample_batch(
                 print(f"{n_samples - len(envs)}/{n_samples} done")
         if train:
             # Compute rewards
-            obs, actions, states, parents, parents_a, done, path_id, state_id = zip(
-                *batch
-            )
+            (
+                obs,
+                actions,
+                states,
+                parents,
+                parents_a,
+                done,
+                path_id,
+                state_id,
+                masks,
+            ) = zip(*batch)
             t0_rewards = time.time()
             rewards = env.reward_batch(states, done)
             t1_rewards = time.time()
             times["rewards"] += t1_rewards - t0_rewards
             rewards = [tf([r]) for r in rewards]
             done = [tl([d]) for d in done]
             batch = list(
-                zip(obs, actions, rewards, parents, parents_a, done, path_id, state_id)
+                zip(
+                    obs,
+                    actions,
+                    rewards,
+                    parents,
+                    parents_a,
+                    done,
+                    path_id,
+                    state_id,
+                    masks,
+                )
             )
         t1_all = time.time()
         times["all"] += t1_all - t0_all
         return batch, times
 
-    def flowmatch_loss(self, it, batch):
+    def flowmatch_loss(self, it, batch, loginf=1000):
```

Comment:
do you think it would be better to make loginf a member variable of the `class GflownetAgent`?